### ISSUE TITLE
nvm_prompt_info doesn't work with Homebrew installed nvm

### ISSUE THREAD

**Issue by jeffcharles** [ohmyzsh/ohmyzsh#4296]
The command `nvm_prompt_info` returns an empty string for me even though I have nvm installed (with Homebrew). It looks to me like there's a bug in lib/nvm.zsh in that it bails out if it can't find nvm at `$HOME/.nvm/nvm.sh` whereas my nvm is installed at `/usr/local/opt/nvm/nvm.sh`. On closer inspection it also looks like plugins/nvm/nvm.plugin.zsh and plugins/nvm/_nvm also make incorrect assumptions about the path where nvm is installed.

**Comment by apjanke** [comment]
Yeah, looks like we should be looking for nvm in a couple more places.

###  Side note on Homebrew `nvm`  ###

It looks like Homebrew's setup instructions for `nvm` are incomplete, or at least of two minds about where the installation should go. It asks you to manually set up that `~/.nvm` directory. But only with `nvm-exec`, and not `nvm.sh`.

```
> $ brew info nvm
nvm: stable 0.26.1 (bottled), HEAD
Manage multiple Node.js versions
https://github.com/creationix/nvm
/usr/local/Cellar/nvm/0.26.1 (6 files, 84K) *
  Poured from bottle
From: https://github.com/Homebrew/homebrew/blob/master/Library/Formula/nvm.rb
==> Caveats
Add NVM's working directory to your $HOME path (if it doesn't exist):

  mkdir ~/.nvm

Copy nvm-exec to NVM's working directory

  cp $(brew --prefix nvm)/nvm-exec ~/.nvm/

Add the following to $HOME/.bashrc, $HOME/.zshrc, or your shell's
equivalent configuration file:

  export NVM_DIR=~/.nvm
  source $(brew --prefix nvm)/nvm.sh

Type `nvm help` for further information.

Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
```

But that `~/.nvm/nvm-exec` doesn't work, because it expects `nvm.sh` to be in the same directory.

```
janke@eilonwy ~                                                      [15:25:52]
> $ ./.nvm/nvm-exec whatever
./.nvm/nvm-exec: line 5: /Users/janke/.nvm/nvm.sh: No such file or directory
./.nvm/nvm-exec: line 10: nvm: command not found
No .nvmrc file found
./.nvm/nvm-exec: line 13: exec: whatever: not found
```

I'm not sure if this means you need to copy over `nvm.sh` as well, or if `nvm` can be run completely from the `$(brew --prefix nvm)` location. Either way, wouldn't be hard to update the OMZ part. And we should report the installation doco error upstream to Homebrew.

Is your NVM installation happy running from `/usr/local/opt`, without any scripts in `~/.nvm`? And is `/usr/local/opt` your `brew --prefix`? It seems to work for me, but I'm no NVM experet.

###  What to do  ###

This is a bit tricky because OMZ support for `nvm` is split between `lib/nvm.zsh` (which provides `nvm_prompt_info`) and `plugins/nvm` (which loads `nvm`). Both are hardcoded to look for NVM in the nvm-default `~/.nvm` location. But, since all `lib/nvm.zsh` provides is a prompt function, and that's not evaluated until after plugins run, it can still be done with the current organization.

I think probably the thing to do is:
* Have `plugins/nvm` look in `~/.nvm`, `$(brew --prefix nvm)` (if present), and `$NVM_DIR` (to allow user to load it from an arbitrary location controlled by `~/.zshrc`) when locating the `nvm` installation. (And maybe on `$PATH`?)
* Have `nvm_prompt_info` and `plugins/nvm/_nvm` look in `$NVM_DIR` instead of `$HOME/.nvm`

Thoughts? /cc @isqua @paulmelnikow, as original authors of the OMZ NVM stuff.

**Comment by jeffcharles** [comment]
I source the nvm shell script in /usr/local/opt in my .profile and zsh profile. It seems to work with that setup.

**Comment by apjanke** [comment]
Sounds consistent. Can you confirm that your `$NVM_DIR` is pointing there after sourcing it? E.g. I get this on OS X 10.9.5 with Homebrew's current `nvm` install.

```
$ source /usr/local/opt/nvm/nvm.sh
$ echo $NVM_DIR
/usr/local/opt/nvm
$
```

**Comment by apjanke** [comment]
I added support for the Homebrew NVM install location in PR #4338. Want to give it a try?

**Comment by jeffcharles** [comment]
I can confirm that $NVM_DIR is pointing at `/usr/local/opt/nvm` after sourcing it. The PR you've referenced seems to fix nvm_prompt_info for me.

**Comment by apjanke** [comment]
Thanks. I'll try to finish testing that PR tonight so we can merge it in soon.

**Comment by mcornella** [comment]
Fixed by #8316.
